### ISSUE TITLE
Pseudo classes and elements broken in recent version

### ISSUE THREAD

**Issue by jaydenseric** [zeit/styled-jsx#219]
Since upgrading `next@3.0.0-beta6` to `next@3.0.0-beta8`, chained pseudo classes such as `:not()` and `:disabled` are broken. I assume this is due to a newer version of styled-jsx. I tried the same styles using https://stylis.js.org without the issue appearing, so it must be something to do with styled-jsx?

With this contrived example:

```jsx
const Textarea = props => (
  <div>
    <textarea {...props} />
    <style jsx>{`
      textarea:not(:focus) {
        border-color: blue;
      }
      textarea:not(:focus):disabled {
        opacity: 0.5;
      }
    `}</style>
  </div>
)
```

Before update (correct):

```css
textarea[data-jsx="451418276"]:not(:focus) {
  border-color: blue;
}

textarea[data-jsx="451418276"]:not(:focus):disabled {
  opacity: 0.5;
}
```

After update (incorrect):

```css
textarea[data-jsx="451418276"]:not([data-jsx="451418276"]:focus) {
  border-color: blue;
}

textarea[data-jsx="451418276"]:not([data-jsx="451418276"]:focus) [data-jsx="451418276"]:disabled {
  opacity: 0.5
}
```

**Comment by jaydenseric** [comment]
- Worked: [styled-jsx@0.5.7](https://github.com/zeit/next.js/blob/3.0.0-beta6/package.json#L90).
- Not working: [styled-jsx@1.0.3](https://github.com/zeit/next.js/blob/3.0.0-beta8/package.json#L89).

@thysultan nothing comes to mind?

**Comment by sidorares** [comment]
having same issue ( hi @jaydenseric ! ) 
`:before` selector not working since updating to next@2.4.1 today

**Comment by thysultan** [comment]
I'll look into this

@sidorares How does your input vs output look like?
@jaydenseric [https://stylis.js.org](https://stylis.js.org) Does not use cascade isolation so it might not show there.

**Comment by thysultan** [comment]
@jaydenseric Is the first one still valid? e.x doesn't `[data-jsx="451418276"]` just increase specificity for the `:focus` pseudo?

```css
textarea[data-jsx="451418276"]:not([data-jsx="451418276"]:focus)
```

**Comment by sidorares** [comment]
@thysultan was: `.submit-status-yes[data-jsx="1547817510"]:before`, after update to next@2.4.1 ( styled-jsx 1.0.3 ) -  `.submit-status-yes:before[data-jsx="1547817510"]`

**Comment by jaydenseric** [comment]
@thysultan No, it is not valid. `:not()` only accepts a single simple selector.

Invalid:

```css
textarea[data-jsx="451418276"]:not([data-jsx="451418276"]:focus) {
  border-color: blue;
}
```

Valid:

```css
textarea[data-jsx="451418276"]:not([data-jsx="451418276"]):not(:focus) {
  border-color: blue;
}
```

**Comment by thysultan** [comment]
>No, it is not valid. :not() only accepts a single simple selector.

Interesting, i think i was working under the assumption that it accepts any number of valid selectors, so that constraint definitely simplifies things 😄 . 

Will push a patch for these.

**Comment by jaydenseric** [comment]
Awesome, looking forward to the patch and getting it pushed all the way through to a Next.js release ASAP, so we can use it!

On the `:not` syntax; even if it were valid, it is still better smaller for bundle size and less confusing inspector style tracing.

**Comment by thysultan** [comment]
@giuseppeg That would make these [tests](https://github.com/thysultan/stylis.js/blob/master/tests/spec.js#L850) invalid right?

```css
a:not( a +b foo:hover :global(marquee) a) {
...
}
```

**Comment by thysultan** [comment]
@jaydenseric This may change with the selector level 4 spec [https://drafts.csswg.org/selectors-4/#negation](https://drafts.csswg.org/selectors-4/#negation) but no browser supports that yet so i don't think we need to worry about it yet.

**Comment by giuseppeg** [comment]
right, @thysultan probably that's my fault. I was just trying to put together odd nested selectors.

When in doubt use the w3c validator https://jigsaw.w3.org/css-validator/validator

**Comment by thysultan** [comment]
patched in v3.0.14

**Comment by krisztianpuska** [comment]
I'm having a similar issue since 2.4.1. Not sure if it's styled-jsx or stylis related.

I get a 
```
 ReferenceError: /pages/index.js?entry: flex is not defined
```

whenever I use [order](https://developer.mozilla.org/en-US/docs/Web/CSS/order) (with or without display: flex):

```js
<style jsx>{`
.hello { order: 1; }
`}</style>
```

**Comment by thysultan** [comment]
@krisztianpuska directly related to [#29](https://github.com/thysultan/stylis.js/pull/29) patched in v3.0.15

**Comment by giuseppeg** [comment]
Fixed in #223
